File: alarm/src/domain/core/rules/SecurityRule.ts
import { TimeSlot } from './TimeSlot.js'
import { SecurityRuleId } from './SecurityRuleId.js'
import { Contact } from '@common/domain/core'

export type SecurityRuleType = 'range' | 'intrusion'

export interface SecurityRule {
  readonly type: SecurityRuleType

  readonly id: SecurityRuleId

  readonly creatorId: string

  readonly activeOn: string

  readonly description: string

  readonly contacts: Contact[]

  readonly enabled: boolean

  readonly validity: TimeSlot
}

File: recognition/app/domain/core/rules/SecurityRule.py
from typing import List, Literal

from app.domain.core.rules import SecurityRuleId, Contact, TimeSlot
from pydantic import BaseModel


class SecurityRule(BaseModel):
    type: Literal["range", "intrusion"]
    id: SecurityRuleId
    creator_id: str
    active_on: str
    description: str
    contacts: List[Contact]
    enabled: bool
    validity: TimeSlot
